examples/controls_example/public/react_controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -335,37 +331,37 @@ export const getOptionsListControlFactory = (
             return;
           }
 
-          const existsSelected = Boolean(existsSelected$.getValue());
-          const selectedOptions = selections$.getValue() ?? [];
+          const existsSelected = Boolean(selections.existsSelected$.getValue());
+          const selectedOptions = selections.selectedOptions$.getValue() ?? [];
           const singleSelect = singleSelect$.getValue();
 
           // the order of these checks matters, so be careful if rearranging them
           const keyAsType = getSelectionAsFieldType(field, key);
           if (key === 'exists-option') {
             // if selecting exists, then deselect everything else
-            existsSelected$.next(!existsSelected);
+            selections.setExistsSelected(!existsSelected);
             if (!existsSelected) {
-              selections$.next([]);
+              selections.setSelectedOptions([]);
               invalidSelections$.next(new Set([]));
             }
           } else if (showOnlySelected || selectedOptions.includes(keyAsType)) {
             componentApi.deselectOption(key);
           } else if (singleSelect) {
             // replace selection
-            selections$.next([keyAsType]);
-            if (existsSelected) existsSelected$.next(false);
+            selections.setSelectedOptions([keyAsType]);
+            if (existsSelected) selections.setExistsSelected(false);
           } else {
             // select option
-            if (!selectedOptions) selections$.next([]);
-            if (existsSelected) existsSelected$.next(false);
-            selections$.next([...selectedOptions, keyAsType]);
+            if (existsSelected) selections.setExistsSelected(false);
+            selections.setSelectedOptions(
+              selectedOptions ? [...selectedOptions, keyAsType] : [keyAsType]
+            );
```

Comment:
As a cleanup, I wonder if we could move the `makeSelection` function into `initializeOptionsListSelections`? Not a blocker, though

Reply:
The problem with moving  `makeSelection` into `initializeOptionsListSelections` is that in initializeOptionsListSelections the subjects are `BehaviorSubject` instead of `PublishingSubjects`. This makes it easy to slip up and call `subject.next` instead of the `setter`. Is it better to co-locate this code in `initializeOptionsListSelections` or have type protections?

Reply:
Ah, that's a fair point - I'm fine leaving it where it is for now for type safety in that case :+1: